src/javascripts/Runners/assetHandlers/scriptFragmentHandler.js

```diff
@@ -67,23 +73,60 @@ ngapp.run(function(mergeAssetService, assetHelpers, pexService, progressLogger,
         pexService.saveScript(script, newPath);
-        xelib.SetValue(a.handle, '', fileName);
+        let oldFileName = xelib.GetValue(a.handle, `Script Fragments\\fileName`);
+        log(`Changing fragment fileName from ${oldFileName} to ${fileName}`);
+        xelib.SetValue(a.handle, 'Script Fragments\\fileName', fileName);
+        let fragmentHandle = xelib.GetElement(a.handle, 'Script Fragments\\Fragments');
+        if (fragmentHandle) {
+            let numberOfFragments = xelib.ElementCount(fragmentHandle);
+            log(`Found ${numberOfFragments} fragments to fix for ${fileName}`);
+            for (let i = 0; i < numberOfFragments; i++) {
+                let oldValue = xelib.GetValue(a.handle, `Script Fragments\\Fragments\\[${i}]\\scriptName`);
+                if (oldValue === oldFileName) { 
+                    log(`Changing fragment ${i} scriptName from ${oldValue} to ${fileName}`);
+                    xelib.SetValue(a.handle, `Script Fragments\\Fragments\\[${i}]\\scriptName`, fileName);
+                } else {
+                    log(`Not changing fragment ${i} scriptName from ${oldValue} to ${fileName} (different script)`);
+                }
+            }
+        }
+        let scriptHandle = xelib.GetElement(a.handle, 'Scripts');
+        if (scriptHandle) {
+            let numberOfScripts = xelib.ElementCount(scriptHandle);
+            log(`Found ${numberOfScripts} scripts to fix for ${fileName}`);
+            for (let i = 0; i < numberOfScripts; i++) {
+                let oldValue = xelib.GetValue(a.handle, `Scripts\\[${i}]\\scriptName`);
+                if (oldValue === oldFileName) {
+                    log(`Changing script ${i} scriptName from ${oldValue} to ${fileName}`);
+                    xelib.SetValue(a.handle, `Scripts\\[${i}]\\scriptName`, fileName);
+                } else {
+                    log(`Not changing script ${i} scriptName from ${oldValue} to ${fileName} (different script)`);
+                }
+            }
+        }
     };
 
     mergeAssetService.addHandler({
         label: 'Script Fragments',
         priority: 1,
         get: function(merge) {
             forEachPlugin(merge, (plugin, folder) => {
-                let assets = findScriptFragments(merge, plugin, folder);
+                let assets = findScriptFragments(merge, plugin, folder, false);
                 if (assets.length === 0) return;
                 merge.scriptFragments.push({ plugin, folder, assets });
             });
         },
         handle: function(merge) {
+            progressLogger.log('Handling Script Fragments');
+            merge.scriptFragments = [];
+            mergedPluginFragments = [];
+            forEachPlugin(merge, (plugin, folder) => {
+                let assets = findScriptFragments(merge, plugin, folder, true);
+                if (assets.length === 0) return;
+                merge.scriptFragments.push({ plugin, folder, assets });
+            });
```

Comment:
while you could re-retrieve the script fragments at this stage, I don't know if it's in line with the intent behind these asset handlers.  I think we would be better served by actually persisting the entries and using information available int he merge to access the correct record from the merge.

Reply:
The problem was I need handles for the merged plugin, not the original ones - I guess I could use the map to find where it should be in the merged plugin but I thought this was safer..?

What would you suggest?

Reply:
I'm not sure.  I think it might be better to actually find the fragments in the merged plugin by mapping from the entries retrieved from the unmerged plugins, but I'll have to make sure that there isn't some circumstance in which that would lead to an undesirable outcome.

Reply:
I think that should be fine - I'll look into the api to see what the best way is